### DISCUSSION TITLE
zh/development/intro

### DISCUSSION THREAD

**Discussion by giscus** [ryan4yin/nixos-and-flakes-book#151] [Announcements]
# zh/development/intro

一份非官方的新手指南

https://nixos-and-flakes.thiscute.world/zh/development/intro

**Comment by jinzhongjia** [comment]
像这样独立出开发环境，那么编辑器的配置呢？
例如我使用的是neovim，它的配置也可以独立出来然后由对应的独立环境引入吗？

**Reply by jinzhongjia** [reply]
因为如果不能把编辑器放到对应项目的开发环境中或者独立出来，那么编辑器的配置就不得不放在home manager中，这样就又导致整个系统出现开发环境的问题，有什么好的解决方案呢
还有就是我有时也会使用vscode来写前端，vscode如何直接自动重载开发环境的flake.nix呢

**Reply by TwIStOy** [reply]
我理解可以wrap一下neovim的命令，替换掉stdpath然后使用本项目环境里的配置文件吧。

**Reply by jinzhongjia** [reply]
不是，我的开发的话比较杂，zig c go py 还有nodejs的也写，我如何在每个项目环境里面只用或者把整个neovim配置引入呢？因为我是最近才迁移到nixos,很多功能都不了解

**Comment by ryan4yin** [comment]
> 我理解可以wrap一下neovim的命令，替换掉stdpath然后使用本项目环境里的配置文件吧。

@jinzhongjia @TwIStOy 
我当初写这部分内容的时候就是这么做的，但后面遇到些问题，最后还是改成了直接用 `home.packages`  装了 neovim 的所有依赖...

这部分内容有点过时了，我会修改一下，每个语言或项目的开发环境中完全只装它自己需要的东西，这是比较理想的，但实际处理起来比较麻烦。

**Reply by jinzhongjia** [reply]
我在看nixvim这个项目，是否可以直接将neovim的配置直接独立出来（例如作为一个单独的flake），然后根据传入参数不同决定启用neovim的部分功能呢

**Reply by jinzhongjia** [reply]
我个人有点强迫症，都用上nixos了，再利用不起来它本身天然的独立构建开发环境的特性有点难受。。。
